File: ConnectFour/app/src/main/java/com/example/connectfour/MainActivity.java
package com.example.connectfour;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    private Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Set up the Toolbar
        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        // Initialize BottomNavigationView
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        // Initialize NavHostFragment
        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager()
                .findFragmentById(R.id.nav_host_fragment);

        if (navHostFragment != null) {
            // Initialize NavController
            NavController navController = navHostFragment.getNavController();

            // Configure the AppBarConfiguration with the correct fragment IDs
            AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                    R.id.boardFragment, R.id.gameOptionsFragment) // Use correct fragment IDs
                    .build();

            // Set up ActionBar with NavController using the Toolbar
            NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);

            // Set up BottomNavigationView with NavController
            NavigationUI.setupWithNavController(bottomNavigationView, navController);

            // Listener to change the toolbar title dynamically
            navController.addOnDestinationChangedListener((controller, destination, arguments) -> {
                if (destination.getId() == R.id.boardFragment) {
                    toolbar.setTitle("Connect Four");
                } else if (destination.getId() == R.id.gameOptionsFragment) {
                    toolbar.setTitle("Options");
                }
            });
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = ((NavHostFragment) getSupportFragmentManager()
                .findFragmentById(R.id.nav_host_fragment)).getNavController();
        return navController.navigateUp() || super.onSupportNavigateUp();
    }
}

File: ConnectFour/app/src/main/java/com/example/connectfour/MainActivity.kt
package com.example.connectfour

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    companion object {
        private const val REQUEST_CODE_OPTIONS = 1 // Request code for starting GameOptions activity
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // Set the content view to your main layout

        // Handle the "Connect Four" button click
        findViewById<View>(R.id.button).setOnClickListener {
            // Start Board activity when Connect Four button is clicked
            val intent = Intent(this@MainActivity, Board::class.java)
            startActivity(intent)
        }

        // Handle the "Options" button click
        findViewById<View>(R.id.button_options).setOnClickListener {
            // Start GameOptions activity and listen for result
            val intent = Intent(this@MainActivity, GameOptions::class.java)
            startActivityForResult(intent, REQUEST_CODE_OPTIONS) // Request code 1 for options
        }
    }

    // Override onActivityResult to handle the result from GameOptions
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        // Check if the result is from GameOptions and if it was successful
        if (requestCode == REQUEST_CODE_OPTIONS && resultCode == RESULT_OK && data != null) {
            // Get the selected mode from the Intent
            val selectedMode = data.getStringExtra("gameMode")

            // Display a Toast message with the selected mode
            Toast.makeText(this, "$selectedMode mode", Toast.LENGTH_SHORT).show()
        }
    }
}
